services/src/Infrastructure/ConfigureServices.cs

```diff
@@ -57,7 +58,9 @@ public static void AddBeforeSaveChangesPipeline(this IServiceCollection services
         services.AddScoped<IAddCreationInfoOperation, AddCreationInfoOperation>();
         services.AddScoped<IAddUpdateInfoOperation, AddUpdateInfoOperation>();
         services.AddScoped<IRemovalHandlingOperation, RemovalHandlingOperation>();
-
+        services.AddHostedService<CheckUserEndBanDate>();

```

Comment:
This is not a correct method for this line of code. This method is intended to initialize db pipeline. Moreover i do not see a reasone to create this service in infrastructure service if you already add dependency for background tasks in application layer. So It could be moved do ConfigureServices in Application layer.